viewing_party/party.py

```diff
@@ -1,23 +1,258 @@
+    
+# removes movies from friends watched list if user has already watched
+    for val in user_watch_list:
+        if val in unique_friend_watch:
+            unique_friend_watch.remove(val)
+            
+    return unique_friend_watch
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
+def get_available_recs(user_data):
+  # variables
+    user_watched_list = user_data["watched"]
+    user_subs_list = user_data["subscriptions"]
+    user_friends_list = user_data["friends"]
+
+  # gets multiple friends watch lists
+    friends_list = []
+    for dicts in user_friends_list:
+        friends_list.append(dicts["watched"])
+
+  # compiles friends watch list into one list
+    friends_rec_list = []
+    for sublist in friends_list:
+        for val in sublist:
+            friends_rec_list.append(val)
+
+  # removes movies that user has already watched from friends recommendations list
+    for dicts in user_watched_list:
+        if dicts in friends_rec_list:
+            friends_rec_list.remove(dicts)
+
+  # retrieves recommendations if user has subscription to site
+    recommendations = []
+    for subs in user_subs_list:
+        for dicts in friends_rec_list:
+            if dicts['host'] == subs:
+                recommendations.append(dicts)
+                break
+
+    return recommendations
 
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+
+  #user data
+    user_watched_list = user_data['watched']
+
+    if not user_watched_list:
+        genre_list = []
+        most_watched_genre = None
+    else:
+        genre_list = [d['genre'] for d in user_watched_list]
+        most_watched_genre = max(set(genre_list), key=genre_list.count)
+ 
+  #friend data
+
+    user_friends_list = user_data["friends"]
+    friends_list = []
+    for dicts in user_friends_list:
+        friends_list.append(dicts["watched"])
+
+    friends_list = []
+    for dicts in user_friends_list:
+        friends_list.append(dicts["watched"])
+
+  # compiles friends watch list into one list
+    friends_rec_list = []
+    for sublist in friends_list:
+        for val in sublist:
+            friends_rec_list.append(val)
+
+  # creates recommendation list by most watched genre
+    rec_by_genre = []
+    for dict in friends_rec_list:
+        if dict['genre'] == most_watched_genre:
+            rec_by_genre.append(dict)
+
+  # removes repeated recommendations
+    recommendations = []
+    for dicts in rec_by_genre:
+        if dicts not in recommendations:
+            recommendations.append(dicts)
+   
+  # removes recommendations if user has already watched movie
+    for dicts in user_watched_list:
+        if dicts in recommendations:
+            recommendations.remove(dicts)
+        
+    return recommendations

```

Comment:
You can also make use of previous functions instead of repeating the same logic. I noticed you are using several loops to capture data from the friends list and user list. You could also simplify this by using nested loops. I'd like to challenge you to refactor this code to using the previous functions and using only one loop. Something like this:
```
def get_new_rec_by_genre(user_data):
    most_freq_genre = get_most_watched_genre(user_data)
    unique_friend_movies = get_friends_unique_watched(user_data)
```